app/forms/agreement_form.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class AgreementForm < Reform::Form
+  property :title, virtual: true
+  property :source_id, virtual: true
+  property :agreement_file, virtual: true
+
+  validates :title, presence: true
+  validates :source_id, format: { with: /\A\w+:\w+\z/,
+                                  message: 'must have a colon' },
```

Comment:
If we want to be precise, the message is technically "must have a single colon in the middle".